engines/terraform/deployment.go

```diff
@@ -73,8 +73,15 @@ func (td *TerraformDeployment) resolveInfraResource(infraName string) (cdktf.Ter
 
 		td.createVariablesForIntent(infraName, resource)
 
+		sugaVar := map[string]any{
+			"stack_id": td.stackId.Result(),
+		}
+
 		td.terraformInfraResources[infraName] = cdktf.NewTerraformHclModule(td.stack, jsii.String(infraName), &cdktf.TerraformHclModuleConfig{
 			Source: jsii.String(pluginRef.Deployment.Terraform),
+			Variables: &map[string]interface{}{
+				"suga": sugaVar,
+			},
 		})
 	}
```

Comment:
Originally the idea behind not including this was to allow any existing terraform module to be made into infra modules without any additional HCL wrapping. As far as unique naming goes at the moment this can be delegated to SREs (or whoever is performing the `apply`) by forwarding the name using variables.

Not saying we don't include this by the way, just sharing the original thinking behind not including `suga` vars for infra.

Reply:
right, that would be the way to do it currently, could potentially use a few variables as well. e.g.
`${var.project_name}-${var.environment_name}-vpc` as an example. Built-in variables are also possible.

e.g. `${var.project_name}-${terraform.workspace}`, although it might be better to default for those rather than make an assumption i.e. `var.environment_name` == `terraform.workspace` as a potential default.